### PR TITLE
[minor] Add @capturesCallingObject tag handling

### PR DESCRIPTION
**Pull request by legutieMSFT**
Thank you for your pull request!  

Please add '[major]', '[minor]', or [patch] to the title to indicate the impact the change has on the code.  Please also provide the following information.

---

**Change Description**:

    Handle @capturesCallingObject metadata tag for Excel javascript custom functions

1. **Do these changes impact *command syntax* of any of the packages?** (e.g., add/remove command, add/remove a command parameter, or update required parameters)
    No

2. **Do these changes impact *documentation*?** (e.g., a tutorial on https://learn.microsoft.com/office/dev/add-ins/overview/office-add-ins)
    Yes, but will be handled separately to indicate what the flag means

If you answered yes to any of these please do the following:
    > Include 'Rick-Kirkham' in the review
    > Make sure the README file is correct

**Validation/testing performed**:

    Will validate it works with scriptlab in excel

### CODE REVIEW

**Comment by legutieMSFT** [comment]
> @microsoft-github-policy-service agree company="Microsoft"

@microsoft-github-policy-service agree company="Microsoft"

**Review by jeaniehuynh — APPROVED** [review]

File: packages/custom-functions-metadata/Specification.md [hunk 1]
```diff
@@ -32,6 +46,13 @@ If the last function parameter is of type `CustomFunctions.CancelableInvocation`
 
  A function cannot have both `@cancelable` and `@streaming` tags.
 
+---
+### @capturesCallingObject
+
+Indicates that the custom function captures the calling object i.e. the [Excel.CellValue](https://learn.microsoft.com/es-es/javascript/api/excel/excel.cellvalue)
+object that invoked the custom function, in the first parameter.
+
+This tag is required to generate metadata for the custom function.
```
Comment (rdong999) [inline packages/custom-functions-metadata/Specification.md]: Please improve the wording for "This tag is required to generate metadata for the custom function."

File: packages/custom-functions-metadata/test/cases/error-linkedentitydataprovider-capturesCallingObject/functions.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+/**
+ * Test the linkedEntityDataProvider tag in combination with the volatile tag
```
Comment (rdong999) [inline packages/custom-functions-metadata/test/cases/error-linkedentitydataprovider-capturesCallingObject/functions.js]: nit: Please change volatile to capturesCallingObject

**Review by rdong999 — APPROVED** [review]

**Review by akrantz — CHANGES_REQUESTED** [review]
Generally this looks good, but made some suggestions about spec updates. I also feel like it should validate that the first parameter type is correct for the calling object.

File: packages/custom-functions-metadata/test/cases/capturesCallingObject/expected.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            "description": "basic custom function with callingObject parameter",
+            "id": "BASIC",
+            "name": "BASIC",
+            "options": {
+                "capturesCallingObject": true
+            },
+            "parameters": [
+                {
+                    "name": "callingObject",
+                    "type": "any"
```
[the remark this answers is not in the document]
Comment (legutieMSFT) [reply packages/custom-functions-metadata/test/cases/capturesCallingObject/expected.json:14]: I'm adding validation for 1 required, scalar and non-repeating parameter. Will push once that is done

**Review by akrantz — DISMISSED** [review]
Looks good now. Thanks for making the updates.

As we discussed, there will be minimum validation for now, and we will discuss about improving that in the future.

**Review by jeaniehuynh — APPROVED** [review]

**Review by rdong999 — APPROVED** [review]

**Review by MiaofeiWang — APPROVED** [review]

**Review by Adrian-MSFT — APPROVED** [review]

**Comment by akrantz** [comment]
I will approve to unblock this PR. There's still an opportunity to help the developer by validating that they have a parameter for the calling object and it is of an appropriate type.

**Review by akrantz — APPROVED** [review]

**Pull request merged**